code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -706,9 +706,9 @@
 /obj/item/borg/upgrade/expand/deactivate(mob/living/silicon/robot/R, user = usr)
 	. = ..()
 	if (.)
-		if (R.hasExpanded)
-			R.hasExpanded = FALSE
+		while(R.expansion_count)
 			R.resize = 0.5
+			R.expansion_count--
 			R.update_transform()
```

Comment:
I'm not 100% a fan of using a while loop here as its technically not needed but I suppose it prevents bugs from turning borgs invisible by rounding their size down to 0